src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift

```diff
@@ -17,18 +17,11 @@ enum Route {
 	}
 
 	init?(url: URL) {
-		#if DEBUG
-		// Dev code!
-		self = .checkin("BAAREMYIAEJBYQLEOZQW4Y3FMQQE22LDOJXSARDFOZUWGZLTFQQES3TDFYNA2OJVEBHGSZLLN4QEYYLOMUUAAMAADJYQQAISLMYFSMATAYDSVBSIZY6QEAIGBAVIMSGOHUBQCBYDIIAAI45QYSZNOFBBC4ZJG7SOMAZH3Z7YKOBXKFMGY2BVZJHUZSXM3BDT26RSZYVEJHHKO4NSGVYNLJND4LO4AH2CNRPLQUYZACC7N65AHFPBUEDW3VYT4OT7FQWBVRW5D643HDA5EIDAQAIQBAMCG")
-		#else
 		let components = URLComponents(url: url, resolvingAgainstBaseURL: true)
-		guard components?.host?.lowercased() == "e.coronawarn.app",
-			let lowercasedPath = components?.path.lowercased(),
-			  lowercasedPath.contains("/c1") else {
+		guard components?.host?.lowercased() == "e.coronawarn.app" else {
 			return nil
 		}
-		self = .checkin(url.lastPathComponent)
-		#endif
+		self = .checkin(url.absoluteString)
```

Comment:
Only verifying the host isn't enough - if you accept everything appended to the URL you can easily start the app and inject some codes. You could at least check for the right query parameter here?

If the Tech Spec changed please let me know where the URL checks are described.

`QRCodeVerificationHelper`should only validate if the content or the QR Code is correct? It's not responsible for the URLs.
Please add the the give MARKS and move test to the designated folder `__tests__`

Reply:
To be done in this subtask: https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6273